test/diff.spec.ts

```diff
@@ -0,0 +1,49 @@
+import diff from '../src/diff';
+import { parser } from '../src/parser';
+
+import {
+  firstDocument,
+  secondDocument,
+  diffLocalOutput,
+  diffOutput,
+} from './fixtures/diff.fixtures';
+
+describe('Diff', () => {
+  test('Check if diff is an empty array for same inputs', () => {
+    expect(diff(firstDocument, firstDocument)).toStrictEqual([]);
+  });
+
+  test('Check diff output with local inputs', () => {
+    expect(diff(firstDocument, secondDocument)).toStrictEqual(diffLocalOutput);
+  });
+
+  test('Check diff output through parser with no difference', async () => {
+    const parsedData = await parser(
+      './test/spec/asyncapi.yml',
+      'https://raw.githubusercontent.com/asyncapi/parser-js/master/test/good/asyncapi-messages-channels.yml'
+    );
+    expect(
+      diff(parsedData.firstDocument.json(), parsedData.secondDocument.json())
+    ).toStrictEqual([]);
+  });
+
+  test('Check diff output through parser with no difference', async () => {
+    const parsedData = await parser(
+      './test/spec/asyncapi.yml',
+      'https://raw.githubusercontent.com/asyncapi/parser-js/master/test/good/asyncapi-messages-channels.yml'
+    );
+    expect(
+      diff(parsedData.firstDocument.json(), parsedData.secondDocument.json())
+    ).toStrictEqual([]);
+  });
```

Comment:
You have duplicated tests, also using the remote doc files isn't good idea, better would be if you save `https://raw.githubusercontent.com/asyncapi/parser-js/master/test/good/asyncapi-messages-channels.yml` doc in the fixtures.

Reply:
I added the remote URL as a way to test if the parser is getting the URL or not. Will move the URL data into a fixture file.